controllers/storagecluster/cephcluster.go

```diff
@@ -1130,6 +1141,49 @@ func createOrUpdatePrometheusRule(r *StorageClusterReconciler, sc *ocsv1.Storage
 	return nil
 }
 
+// setReadAffinity sets read affinity by setting "CSI_ENABLE_READ_AFFINITY"
+// in ocs-operator-config configmap.
+func (r *StorageClusterReconciler) setReadAffinity(sc *ocsv1.StorageCluster) error {
+	var (
+		readAffinityVarKey = "CSI_ENABLE_READ_AFFINITY"
+		readAffinityVarVal = strconv.FormatBool(!sc.Spec.ExternalStorage.Enable)
+		cm                 = &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      util.OcsOperatorConfigName,
+				Namespace: sc.Namespace,
+			},
+			Data: map[string]string{
+				readAffinityVarKey: readAffinityVarVal,
+			},
+		}
+		needsRestart = false
+	)
+
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, cm, func() error {
+		if cm.Data[readAffinityVarKey] != readAffinityVarVal {
+			cm.Data[readAffinityVarKey] = readAffinityVarVal
+			needsRestart = true
+		}
+		return nil
+	})
+	if err != nil {
+		r.Log.Error(err, fmt.Sprintf("failed to update %q configmap", util.OcsOperatorConfigName),
+			"CephCluster", klog.KRef(sc.Namespace, sc.Name))
```

Comment:
```suggestion
			"StorageCluster", klog.KRef(sc.Namespace, sc.Name))
```

Please change this at other places as well.